mirgecom/mpi.py

```diff
@@ -154,3 +156,11 @@ def wrapped_func(*args, **kwargs):
         func(*args, **kwargs)
 
     return wrapped_func
+
+
+def comm_any(comm, local_value):
+    """Return True if *local_value* is True on any rank in *comm*."""
+    reduced_value = np.array([int(local_value)])
+    from mpi4py import MPI
+    comm.Allreduce(MPI.IN_PLACE, reduced_value, op=MPI.MAX)
+    return bool(reduced_value[0])
```

Comment:
This is exactly the facility I was talking about when I said we need the ability to handle rank-specific errors.  An error on any rank can be detected using this snippet.  We should be careful to do collective error detection and handling in other places, too.

Reply:
I'm not sure. I recall we discussed this before, but I don't recall what the resolution was. Currently, my thinking is that we should probably rely on mpi4py to kill the job if one rank exits with an exception.